pkg/server/test/list_users.go

```diff
@@ -0,0 +1,93 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openfga/openfga/pkg/server/commands/listusers"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/stretchr/testify/require"
+)
+
+// setupListUsersBenchmark writes the model and lots of tuples
+func setupListUsersBenchmark(b *testing.B, ds storage.OpenFGADatastore, storeID string) (*openfgav1.AuthorizationModel, int) {
+	b.Helper()
+	// this model exercises many execution paths: union, computed usersets, tuple to userset, and direct relationships.
+	model := testutils.MustTransformDSLToProtoWithID(
+		`model
+			schema 1.1
+		type user
+		type folder
+			relations
+				define viewer: [user]
+		type document
+			relations
+				define viewer: [user]
+				define parent: [folder]
+				define can_view: viewer or viewer from parent`)
+	err := ds.WriteAuthorizationModel(context.Background(), storeID, model)
+	require.NoError(b, err)
+
+	numberOfUsersWithAccess := 0
+	for i := 0; i < 100; i++ {
+		var tuples []*openfgav1.TupleKey
+
+		for j := 0; j < ds.MaxTuplesPerWrite(); j++ {
```

Comment:
What is `j` needed for? 

Did you intent to do `strconv.Itoa(j)` below?

Reply:
🤦‍♀️ yes, thank you

Reply:
Got it 👍  - that wasn't clear to me. Thank you.